docs/DeveloperGuide.md

```diff
@@ -136,6 +136,57 @@ Classes used by multiple components are in the `seedu.mcgymmy.commons` package.
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Find feature
+
+#### Implementation
+
+The find mechanism is facilitated by `ModelManager`. It keeps a filtered list (updated via a specified `Predicate`) of food items,
+stored internally as a `FilteredList<Food>`. Additionally, it implements the following relevant operation:
+
+* `ModelManager#updateFilteredFoodList(Predicate<Food>)` - Filters internal storage via a `Predicate<Food>` specification.
+
+Given below is an example usage scenario and how the find mechanism behaves at each step.
+
+Step 1. The user launches the application after having used it for a while.
+
+Step 2. The user wants to find what he ate for dinner on a certain date (eg. 21-10-2020).
+
+Step 3. The user executes `find -n apple -t fruit`. The `find` command will check if the inputs are valid (eg. invalid date format, empty input etc.),
```

Comment:
This part maybe can go higher level (remove specific details on which methods get run and just say what happens, e.g. "it combines the predicates then filters the list"?

Reply:
Sure, makes sense.